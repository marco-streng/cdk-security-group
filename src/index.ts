import * as ec2 from "@aws-cdk/aws-ec2";

export interface INamedSecurityGroup {
  getSecurityGroupName(): string;
}

export class SecurityGroup extends ec2.SecurityGroup implements INamedSecurityGroup {
  getSecurityGroupName(): string {
    return this.securityGroupName;
  }
}
