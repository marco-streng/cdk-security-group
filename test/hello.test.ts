import { Template } from "@aws-cdk/assertions";
import { Vpc } from "@aws-cdk/aws-ec2";
import { Stack } from "@aws-cdk/core";
import { SecurityGroup } from "../src";

describe("PrivateBucket", () => {
  it("should have named security group", () => {
    // Given
    const stack = new Stack();
    const vpc = new Vpc(stack, "vpc");
    // When
    const securityGroup = new SecurityGroup(stack, "id", {
      vpc: vpc,
      securityGroupName: "pepperize",
    });
    // Then
    const template = Template.fromStack(stack);
    template.hasResource("AWS::EC2::SecurityGroup", {});
    expect(securityGroup.getSecurityGroupName()).toEqual("pepperize");
  });

  it("should match snapshot", () => {
    // Given
    const stack = new Stack();
    const vpc = new Vpc(stack, "vpc");
    // When
    new SecurityGroup(stack, "id", { vpc: vpc, securityGroupName: "pepperize" });
    // Then
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
