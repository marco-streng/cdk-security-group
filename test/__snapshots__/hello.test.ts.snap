// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateBucket should match snapshot 1`] = `
Object {
  "Resources": Object {
    "id4D1900E5": Object {
      "Properties": Object {
        "GroupDescription": "Default/id",
        "GroupName": "pepperize",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "vpcA2121C38": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "vpcIGWE57CBDCA": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "vpcPrivateSubnet1DefaultRoute1AA8E2E5": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "vpcPublicSubnet1NATGateway9C16659E",
        },
        "RouteTableId": Object {
          "Ref": "vpcPrivateSubnet1RouteTableB41A48CC",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPrivateSubnet1RouteTableAssociation67945127": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcPrivateSubnet1RouteTableB41A48CC",
        },
        "SubnetId": Object {
          "Ref": "vpcPrivateSubnet1Subnet934893E8",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPrivateSubnet1RouteTableB41A48CC": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPrivateSubnet1Subnet934893E8": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.128.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcPrivateSubnet2DefaultRouteB0E07F99": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "vpcPublicSubnet2NATGateway9B8AE11A",
        },
        "RouteTableId": Object {
          "Ref": "vpcPrivateSubnet2RouteTable7280F23E",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPrivateSubnet2RouteTable7280F23E": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPrivateSubnet2RouteTableAssociation007E94D3": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcPrivateSubnet2RouteTable7280F23E",
        },
        "SubnetId": Object {
          "Ref": "vpcPrivateSubnet2Subnet7031C2BA",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPrivateSubnet2Subnet7031C2BA": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.192.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcPublicSubnet1DefaultRoute10708846": Object {
      "DependsOn": Array [
        "vpcVPCGW7984C166",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "vpcIGWE57CBDCA",
        },
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet1RouteTable48A2DF9B",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPublicSubnet1EIPDA49DCBE": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "vpcPublicSubnet1NATGateway9C16659E": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "vpcPublicSubnet1EIPDA49DCBE",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "vpcPublicSubnet1Subnet2E65531E",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "vpcPublicSubnet1RouteTable48A2DF9B": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPublicSubnet1RouteTableAssociation5D3F4579": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet1RouteTable48A2DF9B",
        },
        "SubnetId": Object {
          "Ref": "vpcPublicSubnet1Subnet2E65531E",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPublicSubnet1Subnet2E65531E": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcPublicSubnet2DefaultRouteA1EC0F60": Object {
      "DependsOn": Array [
        "vpcVPCGW7984C166",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "vpcIGWE57CBDCA",
        },
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet2RouteTableEB40D4CB",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPublicSubnet2EIP9B3743B1": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "vpcPublicSubnet2NATGateway9B8AE11A": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "vpcPublicSubnet2EIP9B3743B1",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "vpcPublicSubnet2Subnet009B674F",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "vpcPublicSubnet2RouteTableAssociation21F81B59": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet2RouteTableEB40D4CB",
        },
        "SubnetId": Object {
          "Ref": "vpcPublicSubnet2Subnet009B674F",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPublicSubnet2RouteTableEB40D4CB": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPublicSubnet2Subnet009B674F": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.64.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "Default/vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcVPCGW7984C166": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "vpcIGWE57CBDCA",
        },
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
}
`;
